### DISCUSSION TITLE
Show cargo-add snippet along with or in place of manifest snippets for crates

### DISCUSSION THREAD

**Discussion by epage** [rust-lang/crates.io#4980] [Feature Requests]
**Currently**. crates.io has the following interfaces

On https://crates.io/search?q=clap
>  clap v3.2.12 [copy]

On https://crates.io/crates/clap
> Install
>
> Add the following line to your Cargo.toml file:
>
> clap = "3.2.12" [copy]

which adapts with the selected version, like on https://crates.io/crates/clap/3.2.10
> Install
>
> Add the following line to your Cargo.toml file:
>
> clap = "3.2.10" [copy]

**Proposed:** we could alternatively have `cargo add` commands with [cargo-add now in cargo with 1.62](https://github.com/rust-lang/rust/blob/master/RELEASES.md#cargo)

On https://crates.io/crates/clap
> Install
>
> Add the following line to your Cargo.toml file:
>
> $ cargo add clap [copy]

*(elide the version)*

which adapts with the selected version, like on https://crates.io/crates/clap/3.2.10
> Install
>
> Add the following line to your Cargo.toml file:
>
> $ cargo add clap@3.2.10 [copy]

*(explicitly list the version)*

**Questions**
- Should we show both, whether in the short term or long term?
- If we eventually show only cargo-add, how long old does cargo need to be before we do that?
- How should we handle the search page?

**Comment by Turbo87** [comment]
https://github.com/rust-lang/crates.io/pull/5875 added the snippet to the sidebar. we don't have it on the search page, but at the moment I'm not sure we should add more noise to that page.
